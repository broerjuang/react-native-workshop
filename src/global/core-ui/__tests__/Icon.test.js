// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import {Icon} from '../index';
let types = {
  ENTYPO: '500px',
  EVIL_ICONS: 'archive',
  FEATHER: 'activity',
  FONTAWESOME: 'glass',
  FOUNDATION: 'address-book',
  IONICONS: 'ios-add-circle',
  OCTICONS: 'alert',
  MATERIAL_ICONS: '3d-rotation',
  MATERIAL_COMMUNITY_ICONS: 'access-point',
  SIMPLE_LINE_ICONS: 'user',
  ZOCIAL: 'acrobat',
};

describe('Icon should render correctly', () => {
  for (let type of Object.keys(types)) {
    it(`should render ${type} corectly`, () => {
      let component: ReactTestRenderer = renderer.create(
        <Icon name={types[type]} size={16} color="grey" type={type} />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  }
});
