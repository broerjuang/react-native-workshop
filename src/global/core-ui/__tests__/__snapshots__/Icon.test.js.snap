// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly should render ENTYPO corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render EVIL_ICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render FEATHER corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render FONTAWESOME corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render FOUNDATION corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render IONICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render MATERIAL_COMMUNITY_ICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render MATERIAL_ICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render OCTICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render SIMPLE_LINE_ICONS corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;

exports[`Icon should render correctly should render ZOCIAL corectly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
